unstructured/ingest/connector/sharepoint.py

```diff
@@ -0,0 +1,312 @@
+    config: SimpleSharepointConfig
+    tenant: None
+
+    def __init__(self, standard_config: StandardConnectorConfig, config: SimpleSharepointConfig):
+        super().__init__(standard_config, config)
+        self._setup_client()
+
+    @requires_dependencies(["office365"])
+    def _setup_client(self):
+        from office365.runtime.auth.client_credential import ClientCredential
+        from office365.sharepoint.client_context import ClientContext
+
+        if self.config.process_all:
+            parsed_url = urlparse(self.config.site_url)
+            site_hostname = (parsed_url.hostname or "").split(".")
+            tenant_url = site_hostname[0].split("-")
+            if tenant_url[-1] != "admin" or (
+                parsed_url.path is not None and parsed_url.path != "/"
+            ):
+                raise ValueError(
+                    "A site url in the form of https://[tenant]-admin.sharepoint.com \
+                        is required to process all sites within a tenant.",
+                )
+            self.base_site_url = parsed_url._replace(
+                netloc=parsed_url.netloc.replace(site_hostname[0], tenant_url[0]),
+            ).geturl()
+
+        self.client = ClientContext(self.config.site_url).with_credentials(
+            ClientCredential(self.config.client_id, self.config.client_credential),
+        )
+
+    @requires_dependencies(["office365"])
+    def _list_files(self, folder, recursive) -> List["File"]:
+        from office365.runtime.client_request_exception import ClientRequestException
+
+        try:
+            items = folder.expand(["Files", "Folders"]).get().execute_query()
+            files = list(items.files)
+            if not recursive:
+                return files
+            for f in items.itemss:
+                files += self._list_files(f, recursive)
+            return files
+        except ClientRequestException as e:
+            if e.response.status_code != 404:
+                logger.info("Caught an error while processing documents %s", e.response.text)
+            return []
+
+    @requires_dependencies(["office365"])
+    def _list_pages(self, site_client) -> list:
+        from office365.runtime.client_request_exception import ClientRequestException
+
+        try:
+            pages = site_client.site_pages.pages.get().execute_query()
+            pfiles = []
+
+            for page_meta in pages:
+                page_url = page_meta.get_property("Url", None)
+                if page_url is None:
+                    logger.info("Missing site_url. Omitting page... ")
+                    break
+                page_url = f"/{page_url}" if page_url[0] != "/" else page_url
+                file_page = site_client.web.get_file_by_server_relative_path(page_url)
+                site_path = None
+                if (spath := (urlparse(site_client.base_url).path)) and (spath != "/"):
+                    site_path = spath[1:]
```

Comment:
I don't like the use of the name spath here. is there a more accurate name? full_site_path?

Reply:
that works